### PR TITLE
fix: handle nil case

### PR DESCRIPTION
**Pull request by barnabasJ**
Another of my tests failed because nil was returned from `Map.get` above

### CODE REVIEW

File: lib/ash/actions/read/read.ex [hunk 1]
```diff
@@ -609,6 +609,9 @@ defmodule Ash.Actions.Read do
                     end)
                   end)
               end
+
+            nil ->
```
Comment (zachdaniel) [inline lib/ash/actions/read/read.ex:613]: Hmmmmmm how are we even getting to `nil` here... I guess we're unloading a calculation or som thing somewhere?

**Review by zachdaniel — APPROVED** [review]

File: lib/ash/actions/read/read.ex (hunk 1 above)
Comment (barnabasJ) [reply lib/ash/actions/read/read.ex:613]: I haven't had much time to dig deeper. I am still working on upgrading to 3.0, and I'm already over my allotted time budget
